components/RuleCard/index.js

```diff
@@ -0,0 +1,65 @@
+import { Fingerprint } from "lucide-react";
+import styled from "styled-components";
+
+export default function RuleCard({ data }) {
+  console.log(data);
+  const truncateText = (text, maxLength) =>
+    text.length > maxLength ? text.slice(0, maxLength) + "..." : text;
+
+  return (
+    <StyledRuleCard>
+      <IconTextWrapper>
+        <StyledRuleIcon />
+        <div>
+          <StyledRuleDescription>
+            {truncateText(data.description, 30)}
+          </StyledRuleDescription>
+          <StyledRuleDetails>
+            beregeltes Eingabefeld: {data.consequences.object}
+          </StyledRuleDetails>
+        </div>
+      </IconTextWrapper>
+    </StyledRuleCard>
+  );
+}
+
+const StyledRuleCard = styled.div`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
```

Comment:
This won't do anything since you have only one child: the `<IconTextWrapper>`. 
Each time you wrap a div around a div you should consider if one container is enough. In this case you can get rid of one of them.